File: Angular_DungeonPathGame/src/app/components/Room/Room.ts

export class Room {
     levelNo: number;
     roomNo: number;
     conxRm1: number;
     conxRm2: number;
     conxType: string;
     roomID: string;

     constructor(_levelNo: number, _roomNo: number, _conxRm1: number, _conxRm2: number, _conxType: string, _roomID: string) 
          
          {
          this.levelNo = _levelNo;
          this.roomNo = _roomNo;
          this.conxRm1 = _conxRm1;
          this.conxRm2 = _conxRm2;
          this.conxType = _conxType;
          this.roomID = _roomID;
          }

}

File: Investigation/Csharp_PathCalculator/models/room.cs
using System;
using System.Collections.Generic;

namespace PathCalculator
{
    public class room
    {

        // attributes
        public int LevelNo;
        public int RoomNo;
        public int NextRoomNo;
        public int NextRoomNoExtr;


        // default constructor
        public room()
        {
            this.LevelNo = 1;
            this.RoomNo = 1;
            this.NextRoomNo = 1;
            this.NextRoomNoExtr = 1;
        }

        // custom constructor
        public room(int _LevelNo, int _RoomNo, int _NextRoomNo, int _NextRoomNoExtr)
        {
            this.LevelNo = _LevelNo;
            this.RoomNo = _RoomNo;
            this.NextRoomNo = _NextRoomNo;
            this.NextRoomNoExtr = _NextRoomNoExtr;
        }

        public void printroom()
        {
            System.Console.Write($"L{this.LevelNo}/R{this.RoomNo}  ");
        }

        public string returnroom()
        {
            string combine = ($"{this.LevelNo} {this.RoomNo}");
            return combine;
        }





    }
}